src/components/companies/edit/CompanyEditDetails.tsx

```diff
@@ -0,0 +1,340 @@
+import { useState, useEffect } from "react";
+import {
+  updateCompanyIndustry,
+  updateCompanyBaseYear,
+  getIndustryGics,
+} from "@/lib/api";
+import type { CompanyDetails as CompanyDetailsType } from "@/types/company";
+import { IconCheckbox } from "@/components/ui/icon-checkbox";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { Undo2 } from "lucide-react";
+import { useTranslation } from "react-i18next";
+
+export function CompanyEditDetails({
+  company,
+  onSave,
+}: {
+  company: CompanyDetailsType;
+  onSave?: () => void;
+}) {
+  const { t } = useTranslation();
+  const [subIndustryCode, setSubIndustryCode] = useState(
+    company.industry?.industryGics?.subIndustryCode
+      ? String(company.industry.industryGics.subIndustryCode)
+      : "",
+  );
+  const [industryVerified, setIndustryVerified] = useState(
+    !!company.industry?.metadata?.verifiedBy,
+  );
+  const [baseYear, setBaseYear] = useState(company.baseYear?.year || "");
+  const [baseYearVerified, setBaseYearVerified] = useState(
+    !!company.baseYear?.metadata?.verifiedBy,
+  );
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [comment, setComment] = useState("");
+  const [source, setSource] = useState("");
+  const [gicsOptions, setGicsOptions] = useState<any[]>([]);
```

Comment:
I'm just going through to try to remove all `any` types, would be nice if we could have a type for this.